bayesian_torch/layers/variational_layers/linear_variational.py

```diff
@@ -126,28 +130,44 @@ def init_parameters(self):
             self.rho_bias.data.normal_(mean=self.posterior_rho_init[0],
                                        std=0.1)
 
+    def kl_loss(self):
+        sigma_weight = torch.log1p(torch.exp(self.rho_weight))
+        kl = self.kl_div(
+            self.mu_weight,
+            sigma_weight,
+            self.prior_weight_mu,
+            self.prior_weight_sigma)
+        if self.mu_bias is not None:
+            sigma_bias = torch.log1p(torch.exp(self.rho_bias))
+            kl += self.kl_div(self.mu_bias, sigma_bias,
+                              self.prior_bias_mu, self.prior_bias_sigma)
+        return kl
+
     def forward(self, input, return_kl=True):
+        if self.dnn_to_bnn_flag:
+            return_kl = False
         sigma_weight = torch.log1p(torch.exp(self.rho_weight))
         weight = self.mu_weight + \
             (sigma_weight * self.eps_weight.data.normal_())
-        kl_weight = self.kl_div(self.mu_weight, sigma_weight,
-                                self.prior_weight_mu, self.prior_weight_sigma)
+        if return_kl:
+            kl_weight = self.kl_div(self.mu_weight, sigma_weight,
+                                    self.prior_weight_mu, self.prior_weight_sigma)
         bias = None
 
         if self.mu_bias is not None:
             sigma_bias = torch.log1p(torch.exp(self.rho_bias))
             bias = self.mu_bias + (sigma_bias * self.eps_bias.data.normal_())
-            kl_bias = self.kl_div(self.mu_bias, sigma_bias, self.prior_bias_mu,
-                                  self.prior_bias_sigma)
+            if return_kl:
```

Comment:
Hey, I think only computing the kl value if return_kl is set to True might break things if we are plugging bayesian layers on a deterministic model. 

Maybe the best way could be if `self.dnn_to_bnn_flag` is set to True, we set it to null, but not computing it at all might just break everything - and we don't have unit tests yet to ensure things won't break.